crates/bevy_render/src/color/mod.rs

```diff
@@ -1332,38 +1322,49 @@ impl encase::private::CreateFrom for Color {
 
 impl encase::ShaderSize for Color {}
 
-#[derive(Debug, Error)]
+#[derive(Debug, Error, PartialEq, Eq)]
 pub enum HexColorError {
     #[error("Unexpected length of hex string")]
     Length,
-    #[error("Error parsing hex value")]
-    Hex(#[from] hex::FromHexError),
+    #[error("Invalid hex char")]
+    Char(char),
 }
 
-fn decode_rgb(data: &[u8]) -> Result<Color, HexColorError> {
-    let mut buf = [0; 3];
-    match hex::decode_to_slice(data, &mut buf) {
-        Ok(_) => {
-            let r = buf[0] as f32 / 255.0;
-            let g = buf[1] as f32 / 255.0;
-            let b = buf[2] as f32 / 255.0;
-            Ok(Color::rgb(r, g, b))
-        }
-        Err(err) => Err(HexColorError::Hex(err)),
+/// Converts hex bytes to an array of rgb\[a\] components
+///
+/// # Example
+/// For RGB: *b"ffffff" -> [255, 255, 255, ..]
+/// For RGBA: *b"E2E2E2FF" -> [226, 226, 226, 255, ..]
+const fn decode_hex<const N: usize>(mut bytes: [u8; N]) -> Result<[u8; N], HexColorError> {
+    let mut i = 0;
+    while i < bytes.len() {
+        // Convert single hex character to u8
+        let val = match hex_value(bytes[i]) {
+            Ok(val) => val,
+            Err(byte) => return Err(HexColorError::Char(byte as char)),
+        };
+        bytes[i] = val;
+        i += 1;
     }
+    // Modify the original bytes to give an `N / 2` length result
+    i = 0;
+    while i < bytes.len() / 2 {
+        // Convert u8 to r/g/b/a
```

Comment:
I'm just a random dude. Happy to provide feedback if it's asked of me, though. I think the comment could say "pairs of `u8`" or something like that. Capitalize `r`, `g`, `b`, `a` (elsewhere too). Maybe "hex digit" instead of "hex character"? None of these are really necessary; more a matter of style.